### DISCUSSION TITLE
Building a workflow to support scanning documents, photos and pages from books then trimming "resizing" the images to create consistent A4 pages at a nominal 300dpi

### DISCUSSION THREAD

**Discussion by klynchk** [ImageMagick/ImageMagick#2639] [Help]
I've been using IM for a couple of years now. I'm able to make the http://www.fmwconcepts.com/imagemagick/index.php scripts work a treat  

I run IM on Ubuntu Focal Fossa 20.04 

Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP
Delegates (built-in): bzlib djvu fftw fontconfig freetype jbig jng jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib

Somewhere along the line I'm missing some of the key concepts for my workflow

I have a workflow where I typically scan A4 pages at 300dpi usually with an extra 10% border on the platter to allow for sub 5° rotation to align the page correctly - occasionally I'll need to crop the image or trim it  (I use irfanview for that interactive stage) and then create a PDF

The workflow works fine except for one issue which is that my page size is not predictable or even consistent in the PDF.

If I'm scanning Christmas cards which are typically 

A6 | 105 mm x 148 mm | 10.5 cm x 14.8 cm
A5 | 148 mm x 210 mm | 14.8 cm x 21 cm
DL | 99 mm x 210 mm | 9.9 cm x 21 cm
125mm Square | 125 mm x 125 mm | 12.5 cm x 12.5 cm

### 1. Page Layout ### 
I'd like the image to be centre of the page and have a white border to fill out to the A4 page edges for the PDF I prefer that the default page orientation is portrait although for a certain big wide card I'd like to rotate the image 90° to fit it nicely on a landscape page.

So how do I process my pages in IM to be correct for when I convert to PDF and bind into a multipage document.

I understand -page a4 is shorthand to specify the dimensions (widthxheight) of the PostScript page in dots per inch (72DPI) but I want my a4page to be 2480 x 3508 (300DPI) or even 4960 x 7016 (600DPI) - This is mainly to futureproof my scans.

Is there a way of showing in the page command I want A4 at a particular DPI (just to make my code more readable by somone other than me)

### 2. Trim Resize Crop Canvas Scaling commands ###

This is where I get lost. I kind of get the individual commands but I struggle to build a coherent model to make my pages consistent

### 3. Searching the old archives on https://legacy.imagemagick.org/discourse-server/ ###

Google offers me results from there except they are not discoverable. How do I search the old legacy queries?

**Comment by snibgo** [comment]
IM won't build PDF pages that have a central raster image plus a white border. It makes each page from a raster image and nothing else. So if you want a white border, you should first add it eg with "-border" or "-extent", to make an image that is 2480 x 3508 pixels.

I suggest that "-page A4" is not useful for you. With IM v7 You can use %[fx...] expressions in "-page", for example %[fx:210/25.4*300], as A4 is 210mm x 297mm.

**Reply by klynchk** [reply]
Than you so much - that explains why when I've used IM to split pdf files 

`convert` "$FILE"  -crop 50%x100% +repage `"$DIRECTORY"/"${FILE%02d.pdf}"`

I've lost the text and it's just been flattened to a raster.
